### ISSUE TITLE
Is there a way to get the ISO-639-2 3-letter code given the language name?

### ISSUE THREAD

**Issue by justhalf** [LuminosoInsight/langcodes#27]
It seems that the library can only produce 2-letter code. Is there a way to map a language into its ISO-639-2 3-letter code?

**Comment by rspeer** [comment]
Hm, I didn't implement that.

I think it'd be a little messy to support outputting codes in the ISO standards that BCP 47 supersedes, because there's more than one of them. For example, if you look up the language name "German", should you get `deu` or `ger`? Should French give you `fra` or `fre`?

**Comment by justhalf** [comment]
Thanks Robyn for the quick response.

I believe that is an issue that is easily resolved, since it is not a real ambiguity, but simply language codes that are associated with different parameters, which could be passed on to the function as well.

In ISO-639-2 case, the only case where there are multiple versions of the same code (for example, "sqi" vs "alb" for Albanian), they are nicely marked whether they are Bibliographic (alb) or Terminology (sqi). (and note that I believe officially the Terminology version is the preferred one, but that doesn't really matter).

For example, I imagine it could be like this:

    >>> german = langcodes.find('German')
    >>> print(german)
    Language.make(language='de')
    >>> print(german.to_tag(standard='ISO-639-2', variant='T'))
    'deu'
    >>> print(german.to_tag(standard='ISO-639-2', variant='B'))
    'ger'
    >>> print(german.to_tag(standard='ISO-639-3'))
    'deu'

(alternatively, we can make separate function for each standard, such as `to_iso639_2_tag` but I feel this is quite ugly)

And even if it is too cumbersome to implement the additional parameters, at least we can add ISO-639-3, which does not have multiple codes for each language.

I felt this is an important feature, as this package is supposed to be solving the boring part of language codes, and getting language code from the language name is one of the important ones (and at least for me it is the most important one, and was surprised that this package does not support that, as I came into this package looking for just that).

I really hope that you would consider implementing this. I would be happy to help, too, if needed.

**Comment by rspeer** [comment]
Actually I do like that design. I won't have time to get to it soon, but I would happily accept a pull request with it.

**Comment by justhalf** [comment]
Ok, noted.

While making pull request, do you have a place for unit tests here?

**Comment by rspeer** [comment]
The code is tested with `pytest --doctest-modules`. Most of the tests are in the docstrings. There is also one standalone test module in `langcodes/tests`.

**Comment by ozburo** [comment]
Just want to comment and second this motion -- we are currently having to add another dependency `pycountry` to get these "ISO 639" 3-letter codes and it would be nice to reduce our dependency coverage.

FWIW: FFMpeg uses ISO 639 for it's subtitle language designation, which is one of our use-cases.

**Comment by rspeer** [comment]
Okay - does https://github.com/LuminosoInsight/langcodes/pull/39/files look like it works the way you want?

I didn't end up using a parameter for the 'standard', because the different standards within ISO 639 mostly serve different purposes. I focused on ISO 639-2 here, because it sounds like that's what we need for interoperability with ffmpeg. ISO 639-3 is just the IANA language tag registry we already use but with less information.

**Comment by rspeer** [comment]
Edge cases I ran into, and as usual they arise from the interaction between Balkan conflicts and standards processes:

- If you ask for the alpha3 bibliographic code of Croatian (`hr`), you'll get `hrv`, the same as the terminology code. It briefly had a bibliographic code of `scc` (which we can recognize and normalize to `hr`), but that code was quickly withdrawn from the standard. Serbian is similar: you get `srp`, not the briefly-included bibliographic code `scr`.

- `sh` gets the alpha3 code of `hbs`. ISO 639-2 says that `sh` doesn't have an alpha3 code, but then it would be the only alpha2 code without an alpha3 code. Meanwhile, CLDR recognizes `hbs` as a 3-letter alias, so I used that. If I followed the ISO standard exactly it'd have to raise an error, and I think that would be an annoying exception.
